### ISSUE TITLE
Nested inlines with default values not being saved.

### ISSUE THREAD

**Issue by mdogo** [datahub/grappelli-nested-inlines#4]
This is an obscure bug so I will give the steps to reproduce the bug.

Given a models structure like this,

``` python
class Root(models.Model):
    # and some fields

class NestedNode(models.Model):
    root = models.ForeignKey('Root')
    # and some fields

class LeafNode(models.Model):
    nested_node = models.ForeignKey('NestedNode')
    # and some fields
```

And the corresponding nested admin,

``` python
class LeafNodeInline(NestedTabularInline):
    model = models.LeafNode

class NestedNodeInline(NestedStackedInline):
    model = models.NestedNode
    inlines = [LeafNodeInline, ]

class RootNodeAdmin(NestedModelAdmin):
    model = models.RootParent
    inlines = [NestedNodeInline, ]
```

When you add a new **NestedNode** with some **LeafNodes** but keeping the default values for the **NestedNode** fields the **NestedNode** itself will not be saved. This is because when the formset for **NestedNode** is saved, it checks all its forms for changes. As the forms has not changed (it has the default values) the **NestedNode** is not created and keeps the **LeafNodes** to be saved too.

I wrote a workaround using my own formset but the solution is not pretty. My code is more or less like this,

``` python
class DefaultSaveInlineFormSet(BaseInlineFormSet):
    def save_new_objects(self, commit=True):
        self.new_objects = []
        for form in self.extra_forms:
            if self.can_delete and self._should_delete_form(form):
                continue
            self.new_objects.append(self.save_new(form, commit=commit))
            if not commit:
                self.saved_forms.append(form)
        return self.new_objects
```

What I do is to always save the forms in the formset, even when they have not changed. This solve the problem with the first time you save the nested object but it will keep saving the nested objects every time you save the root object.

**Comment by allanjamesvestal** [comment]
Thanks for submitting this.

I'm using my own formset in the main project where I'm utilizing this library, so that may have been why I didn't encounter this problem earlier.

 I'll look at the issue and your pull request (probably in the next 24 hours) and will let you know how I plan to solve this. It'll likely incorporate most (if not all) the code you've submitted.

Thanks again for the bug report.

~Allan

**Comment by mdogo** [comment]
I have sent you a pull request that fixes the problem. I did not like the first solution I gave you because that makes the nested instance to be saved every time you save the container instance, including on updates. My second approach was to make the _has_changed_ method of **BaseNestedModelForm** checks if nested formset has changed too but only on creation to prevent saving the instance every time you save the main form.

While I was trying to solve the problem I found that the default form of the **NestedInlineModelAdmin** is not a subclass of **NestedModelFormMixin** so I changed it to use that form by default.

I hope you will find my code useful to solve the problem!

Manuel
